mdformat_myst/plugin.py

```diff
@@ -90,16 +91,20 @@ def _footnote_ref_renderer(node: RenderTreeNode, context: RenderContext) -> str:
 
 
 def _footnote_renderer(node: RenderTreeNode, context: RenderContext) -> str:
-    text = f"[^{node.meta['label']}]: "
-    child_iterator = iter(node.children)
-    first_child = next(child_iterator)
-    if first_child.type == "footnote_anchor":
-        return text
+    first_line = f"[^{node.meta['label']}]:"
+    elements = []
+    for child in node.children:
+        if child.type == "footnote_anchor":
+            continue
+        elements.append(child.render(context))
+    body = indent("\n\n".join(elements), " " * 4)
+    # if the first body element is a paragraph, we can start on the first line,
+    # otherwise we start on the second line
+    if body and node.children and node.children[0].type != "paragraph":
```

Comment:
the `body` condition makes me think, what happens if `not body`. If that is valid syntax, I think it needs separate handling where no trailing space is added?

Reply:
yes it is valid syntax, but yeh I can remove the trailing space. In fact it was removing it, but then I refactored the code slightly and forgot to "rehandle" it

Reply:
actually it only works currently if there is a space, otherwise it is not recognized as a footnote.

I note though, that this is recognized as a footnote in VS Code previews (which uses markdown-it), so perhaps this is a bug in the plugin (albeit a quite unlikely scenario, that you would want a footnote with no text)